src/makeCalculator.js

```diff
@@ -4,7 +4,43 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      if (calculate.hasOwnProperty(callback) === false) {
+        throw new Error('Invalid operation');
+      }
+
+      callback.call(this, number);
+
+      return this;
+    },
+  };
+
+  return calculate;
 }
```

Comment:
The `makeCalculator` function is well-structured and mostly follows the task requirements. However, there are a few issues that need to be addressed to ensure it works as expected.